File: 1442-count-triplets-that-can-form-two-arrays-of-equal-xor/1442-count-triplets-that-can-form-two-arrays-of-equal-xor.cpp
class Solution {
public:
    int countTriplets(vector<int>& arr) {
        int n = arr.size();
        vector<int> prefixXOR(n + 1, 0);
        for (int i = 0; i < n; ++i) {
            prefixXOR[i + 1] = prefixXOR[i] ^ arr[i];
        }

        int count = 0;
        for (int j = 0; j < n; ++j) {
            for (int k = j + 1; k < n; ++k) {
                if (prefixXOR[j] == prefixXOR[k + 1]) {
                    count += (k - j);
                }
            }
        }

        return count;
    }
};

File: 1442-count-triplets-that-can-form-two-arrays-of-equal-xor/1442-count-triplets-that-can-form-two-arrays-of-equal-xor.py
class Solution(object):
    def countTriplets(self, arr):
        prefixXOR = [0] * (len(arr) + 1)
        for i in range(len(arr)):
            prefixXOR[i + 1] = prefixXOR[i] ^ arr[i]

        count = 0
        for j in range(len(arr)):
            for k in range(j + 1, len(arr)):
                if prefixXOR[j] == prefixXOR[k + 1]:
                    count += (k - j)

        return count
        